airbyte/_processors/sql/snowflakecortex.py

```diff
@@ -0,0 +1,257 @@
+    from sqlalchemy.engine import Connection
+
+
+class SnowflakeCortexTypeConverter(SnowflakeTypeConverter):
+    """A class to convert array type into vector."""
+
+    def __init__(
+        self,
+        conversion_map: dict | None = None,
+        *,
+        vector_length: int,
+    ) -> None:
+        self.vector_length = vector_length
+        super().__init__(conversion_map)
+
+    @overrides
+    def to_sql_type(
+        self,
+        json_schema_property_def: dict[str, str | dict | list],
+    ) -> sqlalchemy.types.TypeEngine:
+        """Convert a value to a SQL type."""
+        sql_type = super().to_sql_type(json_schema_property_def)
+        if isinstance(sql_type, sqlalchemy.types.ARRAY):
+            return f"Vector(Float, {self.vector_length})"
+
+        return sql_type
+
+
+class SnowflakeCortexSqlProcessor(SnowflakeSqlProcessor):
+    """A Snowflake implementation for use with Cortex functions."""
+
+    type_converter_class: SnowflakeCortexTypeConverter
+
+    def __init__(
+        self,
+        cache: CacheBase,
+        catalog: ConfiguredAirbyteCatalog | None,
+        vector_length: int,
+        file_writer: FileWriterBase | None = None,
+    ) -> None:
+        """Custom initialization: Initialize type_converter with vector_length."""
+        self._catalog = catalog
+        self._vector_length = vector_length
+        self._engine: Engine | None = None
+        self._connection_to_reuse: Connection | None = None
+        # call base call to do necessary initialization
+        RecordProcessor.__init__(self, cache=cache, catalog_manager=None)
+        self._ensure_schema_exists()
+        self.file_writer = file_writer or self.file_writer_class(cache)
+        self.type_converter = SnowflakeCortexTypeConverter(vector_length=vector_length)
+        self._cached_table_definitions: dict[str, sqlalchemy.Table] = {}
+
+    @overrides
+    def process_airbyte_messages(
+        self,
+        messages: Iterable[AirbyteMessage],
+        write_strategy: WriteStrategy,
+    ) -> None:
+        """Process a stream of Airbyte messages."""
+        if not isinstance(write_strategy, WriteStrategy):
+            raise exc.AirbyteInternalError(
+                message="Invalid `write_strategy` argument. Expected instance of WriteStrategy.",
+                context={"write_strategy": write_strategy},
+            )
+
+        if not self._catalog:
+            raise exc.AirbyteLibInternalError(
+                message="Catalog should exist but does not.",
+            )
+
+        stream_schemas: dict[str, dict] = {}
+
+        # Process messages, writing to batches as we go
+        for message in messages:
+            if message.type is Type.RECORD:
+                record_msg = cast(AirbyteRecordMessage, message.record)
+                stream_name = record_msg.stream
+
+                if stream_name not in stream_schemas:
+                    stream_schemas[stream_name] = self._get_json_schema_from_catalog(
```

Comment:
the base class is calling self.cache.processor.get_json_schema, which is null in this case, so created a local method.